bin/fin

```diff
@@ -3066,6 +3066,40 @@ system_stop ()
 	fi
 }
 
+# Helper function to run ssh-add inside the ssh-agent container
+# @param $1 -D, -l or path to custom key
+_ssh_add ()
+{
+	# Return if no key or option passed
+	[[ "$@" == "" ]] && return
+
+	if is_tty; then
+		# Run ssh-add interactively to allow entering a passphrase for ssh keys (if set)
+		${winpty} docker exec -it docksal-ssh-agent /run.sh ssh-add "$@"
+	else
+		# In a non-interactive environment (e.g., CI) run non-interactively. Keys with passphrases cannot be used!
+		echo-yellow "Running in a non-interactive environment. SSH keys with passphrases cannot be used."
+		docker exec docksal-ssh-agent /run.sh ssh-add "$@"
+	fi
+}
+
+# Helper function to check whether an key is already loaded in the ssh-agent
+# This is useful for keys with a passphrase, which otherwise would require the user to re-enter it.
+# @param $1 host path to the ssh key
+_ssh_key_exists ()
+{
+	# Get fingerprints for keys already loaded in the agent
+	# This function may be called multiple times, so we should cache this value
+	if [[ "$existing_fingerprints" == "" ]]; then
+		existing_fingerprints=$(_ssh_add -l | awk '{print $2}')
+		export existing_fingerprints
+	fi
+
+	new_fingerprint=$(ssh-keygen -lf ${1} | awk '{print $2}')
```

Comment:
Wouldn't just checking by file name be enough?

Reply:
It would, unless the ssh key is regenerated/replaced (same name, different key).

Reply:
I'm just worried about the time it takes to execute all this. How long does it take for say 3 keys?